crates/core/tedge/src/cli/mqtt/error.rs

```diff
@@ -14,4 +16,57 @@ pub enum MqttError {
 
     #[error("MQTT connection error: {0}\n\nHint: Is MQTT server running?")]
     ServerConnection(String),
+
+    #[error("Failed to create a TLS config")]
+    TlsConfig(#[from] CertificateError),
+
+    #[error("Server TLS certificate could not be verified")]
+    ServerCertVerify(#[from] ServerCertVerifyError),
+}
+
+impl From<rumqttc::TlsError> for MqttError {
+    fn from(error: rumqttc::TlsError) -> Self {
+        Self::ServerCertVerify(ServerCertVerifyError::from(error))
+    }
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum ServerCertVerifyError {
+    /// Received certificate is not in X509v3 format
+    #[error("Received certificate is not in X.509 v3 format")]
```

Comment:
What do you mean by received certificate?

```suggestion
    #[error("The MQTT server certificate is not in X.509 v3 format - as required by thin-edge")]
```

Reply:
Reworded this error message as well.